spec/forms/concerns/wizard_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'rails_helper'
+RSpec.describe 'Wizard' do
+  let(:test_wizard_form) do
+    Class.new do
+      include ActiveModel::Model
+      include Wizard
+
+      attr_accessor :name, :email, :phone_number, :other_property
+
+      validates :name, :email, presence: true
+      validates :phone_number, presence: true, on: :new
+    end
+  end
+
+  let(:described_class) { TestWizardForm }
+  let(:store) { instance_double(WizardStateStores::RedisStore, read: nil, delete: nil) }
+  let(:wizard) { described_class.new(store, name: 'Jean', email: 'jean@dfe.com') }
+
+  before do
+    stub_const('TestWizardForm', test_wizard_form)
+  end
+
+  describe '#initialize' do
+    let(:params) { { name: 'Fiona', other_property: '_to_remove' } }
+
+    it 'merges any attributes with ones stored in the data store' do
+      allow(store).to receive(:read).and_return({ name: 'Jane', email: 'fiona@dfe.co.uk' }.to_json)
+
+      wizard = described_class.new(store, params)
+
+      expect(wizard.name).to eq('Fiona')
+      expect(wizard.email).to eq('fiona@dfe.co.uk')
+      expect(wizard.other_property).to eq('_to_remove')
+    end
+
+    context 'when parameter sanitisation is required' do
+      before do
+        described_class.send(:define_method, :sanitize_attributes) do |params|
+          params.delete(:other_property)
```

Comment:
This spec is a little confusing because it tests logic that it implements itself. A check that the sanitize_attributes method is called would be equivalent and simpler.

Reply:
I don't think there is any way I can do that as I can't setup an expectation on an instance of a class without initialising it first. However I have simplified the spec so it's easier to understand.